AskQ/Controllers/UserController.cs

```diff
@@ -11,32 +14,45 @@ namespace AskQ.Controllers
     public class UserController : Controller
     {
         private readonly ApplicationDbContext _dbContext;
+        private readonly UserManager<IdentityUser> _userManager;
 
-        public UserController(ApplicationDbContext dbContext)
+        public UserController(ApplicationDbContext dbContext, UserManager<IdentityUser> userManager)
         {
             _dbContext = dbContext;
+            _userManager = userManager;
         }
 
         // GET /username
         [HttpGet("{username}")]
         public async Task<IActionResult> UserProfileAsync(string username, int page = 1)
         {
-            IdentityUser user = await _dbContext.Users.FirstOrDefaultAsync(u => u.UserName == username);
+            IdentityUser? user = await _userManager.FindByNameAsync(username);
             if (user is null)
             {
                 return NotFound();
             }
-            var viewModel = new UserProfileViewModel
-            {
-                PageNumber = page,
-                User = user,
-                Questions = _dbContext.Questions.Include(q => q.Replies)
-                    .Include(q => q.AskedTo)
-                    .Where(q => q.AskedTo.UserName == username && q.Replies.Any())
+
+            IEnumerable<QuestionViewModel> questions = _dbContext.Questions.Include(q => q.Replies)
+                    .Where(q => q.AskedToGuid == user.Id && q.Replies.Any())
                     .Skip(10 * (page - 1))
                     .Take(10)
                     .OrderByDescending(q => q.DateTime)
-                    .AsEnumerable()
+                    .Select(q => new QuestionViewModel
+                    {
+                        Id = q.Id,
+                        QuestionText = q.QuestionText,
+                        AskedFromUsername = (_userManager.FindByIdAsync(q.AskedFromGuid)).Result.UserName,
+                        DateTime = q.DateTime,
+                        Replies = q.Replies.Select(r => r.ReplyText)
+                    }).AsEnumerable(); // Looks too long.
+                                       // There could be a better approach to populate IEnumerable<QuestionViewModel> from the IEnumerable<Question>;
```

Comment:
@fiseni, this part has two problems.

1. It's noisy a little bit in the code, there might be a better way.
2. It actually doesn't work, "`NullReferenceException: Object reference not set to an instance of an object.`" is thrown in the *UserProfile.cshtml* in the line: `@if (Model.Questions is null || !Model.Questions.Any())`, which is strange to throw null exception because I check already for null. The exact problematic part of this chunk of code is line 44, if I put `AskedFromUsername = "hard-coded string"` it works.

I hope you have the time to review this. Thanks in advance.